config/gradle/common.gradle

```diff
@@ -0,0 +1,119 @@
+apply plugin: 'idea'
+apply plugin: 'groovy'
+apply plugin: 'java'
+
+idea.project?.vcs = 'Git'
+
+// --- Setup script specific properties ---
+
+ext {
+    /* optional - set variable enforceJavaVersion */
+    enforceJavaVersionFlag = project.getProperties().containsKey('enforceJavaVersion')
+    javaVersion = project.getProperties().get('enforceJavaVersion') ?: JavaVersion.VERSION_1_6
+    compileTasks = [compileJava, compileTestJava, compileGroovy, compileTestGroovy]
+}
+
+// --- Setup Global Settings ---
+
+allprojects {
+
+    sourceCompatibility = javaVersion
+    targetCompatibility = javaVersion
+
+    repositories {
+        mavenLocal()
+        mavenCentral()
+    }
+
+    ext {
+        groovyVersion = '2.4.4'
+        customJvmArgs = ['-Xmx1g', '-XX:MaxPermSize=1g']
+    }
+
+    dependencies {
+        testCompile "junit:junit:4.12"
+    }
+
+    configurations {
+        providedCompile
+    }
+
+    sourceSets.main.compileClasspath += configurations.providedCompile
+    sourceSets.test.compileClasspath += configurations.providedCompile
+    sourceSets.test.runtimeClasspath += configurations.providedCompile
+}
+
+// --- Customise JVM - Low Memory Profile ---
+
+task wrapper(type: Wrapper) {
+    gradleVersion = '2.13'
+
+    def jvmOpts = customJvmArgs.join(' ')
+    inputs.property("jvmOpts", jvmOpts)
+    doLast {
+        def optsEnvVar = "DEFAULT_JVM_OPTS"
+        scriptFile.write scriptFile.text.replace("$optsEnvVar=\"\"", "$optsEnvVar=\"$jvmOpts\"")
+        batchScript.write batchScript.text.replace("set $optsEnvVar=", "set $optsEnvVar=$jvmOpts")
+    }
+}
+
+allprojects {
+    clean {
+        delete "out"
+        delete ".gradle"
+    }
+
+    tasks.matching { it instanceof JavaCompile || it instanceof GroovyCompile }.all {
+        options.compilerArgs += ["-Xlint:unchecked", "-Xlint:deprecation"]
+        options.fork = true
+        options.forkOptions.jvmArgs += customJvmArgs
+    }
+
+//    tasks.withType(JavaCompile) {
```

Comment:
Some debug code? I think it's better to remove commented stuff before publishing.